app/elements/app-element.html

```diff
@@ -202,26 +99,27 @@ <h2 class="paper-font-display2">You are not authorized to see this page.</h2>
 			},
 			logout: function() {
 				localStorage.clear();
-				this.refreshMenu(false);
+				this.refreshMenu();
 				page.redirect('/login');
 			},
-			refreshMenu: function(loggedIn) {
-				this.isCompany = !!util.organization();
-				this.isCoder = !this.isCompany;
-				this.isLoggedIn = loggedIn;
+			refreshMenu: function() {
+				this.$.appNav.refresh();
 			},
 			displayInstalledToast: function() {
 				console.log('Caching complete.');
 			},
-			createChallenge: function() {
-				this.$.taskList.createChallenge();
-			},
 			inviteByToken: function(token){
-				this.$.invitationRequest.url = util.build('/invite/auth/' + token);
-				this.$.invitationRequest.generateRequest();
-			},
-			onInvitationResponse: function(r){
-				page('/challenge/' + r.detail.detail.response);
+				var invitationRequest = document.createElement('iron-ajax');
```

Comment:
Using XHR2 (which has onerror and onload) should be no big deal. See http://caniuse.com/#feat=xhr2